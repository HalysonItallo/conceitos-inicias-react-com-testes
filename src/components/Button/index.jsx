import './styles.css'

export function Button({ text, onClick, disabled }) {
  return (
    <button
      className="button"
      type="button"
      onClick={onClick}
      disabled={disabled}
    >
      {text}
    </button>
  )
}
