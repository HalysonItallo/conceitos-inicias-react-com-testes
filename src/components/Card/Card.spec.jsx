import { render, screen } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'
import { Card } from '.'
import { Button } from '../Button'
import { cardPropsMock } from './mock'

describe('<Card />', () => {
  it('should render Card correctly', async () => {
    render(<Card {...cardPropsMock} />)

    expect(screen.getByRole('img', { name: 'title-1' })).toHaveAttribute(
      'src',
      'img/img.png'
    )

    expect(
      screen.getByRole('heading', { name: /title-1/i })
    ).toBeInTheDocument()

    expect(screen.getByText('body-1')).toBeInTheDocument()
  })

  it('should match snapshot', async () => {
    const fn = vi.fn()

    const { container } = render(
      <Button text="Load more" disabled={false} onClick={fn} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
