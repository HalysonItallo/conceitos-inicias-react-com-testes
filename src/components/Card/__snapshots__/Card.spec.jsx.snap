// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Card /> > should match snapshot 1`] = `
<button
  class="button"
  type="button"
>
  Load more
</button>
`;
